java-squid/src/main/java/org/sonar/java/se/symbolicvalues/BinaryRelation.java

```diff
@@ -93,6 +93,23 @@ protected RelationState resolveState(Collection<BinaryRelation> knownRelations)
 
   @CheckForNull
   protected RelationState resolveState(Collection<BinaryRelation> knownRelations, Set<BinaryRelation> usedRelations) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "resolveState" is 21 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)